### ISSUE TITLE
BP node allowed-connection=specified, peer-key= full node pubkey causing nodeos frozen

### ISSUE THREAD

**Issue by jingfei05** [EOSIO/eos#3719]
We have two nodes:
A - full node without producer plugin enabled but private-key is provided.
  #plugin = eosio::producer_plugin
    private-key = [xxxx,xxx]
    #producer-name = eosninecatst
B - BP node use A as the p2p connection and 
allowed-connection=specified
peer-key = "B's public key".

A is working fine but when B is starting up, it is frozen when try to sync data from A.
The following is the log:

OK here is an issue report: configuring my bp behind my full node it's kind of slow. Then when trying to sync data. The nodeos is dead and kill pid will not work in this case.

940566ms thread-0   net_plugin.cpp:1915           connect              ] host: dev.cryptolions.io port: 19876
940737ms thread-0   net_plugin.cpp:934            send_handshake       ] Sending handshake generation 1 to dev.cryptolions.io:19876
940841ms thread-0   net_plugin.cpp:2159           handle_message       ] got a handshake_message from dev.cryptolions.io:19876 dev.cryptolions.io:19876 - f447f3a
940841ms thread-0   net_plugin.cpp:2179           handle_message       ] checking for duplicate
940841ms thread-0   net_plugin.cpp:2668           authenticate_peer    ] Peer dev.cryptolions.io:19876 - f447f3a sent a handshake with an unauthorized key: EOS1111111111111111111111111111111114T1Anm.
940842ms thread-0   net_plugin.cpp:2227           handle_message       ] Peer not authenticated.  Closing connection.
940842ms thread-0   net_plugin.cpp:1094           operator()           ] sent a go away message: authentication failure, closing connection to dev.cryptolions.io:19876
940842ms thread-0   net_plugin.cpp:772            close                ] canceling wait on dev.cryptolions.io:19876
940842ms thread-0   net_plugin.cpp:2078           operator()           ] Error reading message from dev.cryptolions.io:19876: Operation canceled
940842ms thread-0   net_plugin.cpp:772            close                ] canceling wait on dev.cryptolions.io:19876
970566ms thread-0   net_plugin.cpp:1915           connect              ] host: dev.cryptolions.io port: 19876
970659ms thread-0   net_plugin.cpp:934            send_handshake       ] Sending handshake generation 1 to dev.cryptolions.io:19876
970954ms thread-0   net_plugin.cpp:2159           handle_message       ] got a handshake_message from dev.cryptolions.io:19876 dev.cryptolions.io:19876 - f447f3a
970955ms thread-0   net_plugin.cpp:2179           handle_message       ] checking for duplicate
970955ms thread-0   net_plugin.cpp:2668           authenticate_peer    ] Peer dev.cryptolions.io:19876 - f447f3a sent a handshake with an unauthorized key: EOS1111111111111111111111111111111114T1Anm.
970955ms thread-0   net_plugin.cpp:2227           handle_message       ] Peer not authenticated.  Closing connection.
970956ms thread-0   net_plugin.cpp:1094           operator()           ] sent a go away message: authentication failure, closing connection to dev.cryptolions.io:19876
970956ms thread-0   net_plugin.cpp:772            close                ] canceling wait on dev.cryptolions.io:19876
970956ms thread-0   net_plugin.cpp:2312           handle_message       ] got a notice_message from dev.cryptolions.io:19876
970956ms thread-0   net_plugin.cpp:2317           handle_message       ] this is a last irreversible notice with 851085 blocks
970956ms thread-0   net_plugin.cpp:2347           handle_message       ] this is a last irreversible notice with 851419 blocks
970957ms thread-0   net_plugin.cpp:1521           recv_notice          ] sync_manager got last irreversible block notice
970957ms thread-0   net_plugin.cpp:1285           sync_required        ] last req = 0, last recv = 1 known = 851419 our head = 851036
970957ms thread-0   net_plugin.cpp:1405           start_sync           ] Catching up with chain, our last req is 0, theirs is 851419 peer dev.cryptolions.io:19876
^C

Then I will need to kill -9 for nodeo, and this is guaranteed to have a dirty DB flag because of no way to shutdown cleanly.

All dockers OS: Ubuntu 16.04 LTS
A is running in a docker
B is running in a docker as well.

**Comment by halsaphi** [comment]
Please try latest code and documentation. 
https://github.com/EOSIO
https://developers.eos.io/
If problem persists please raise new issue.
